app/assets/javascripts/initializers/initializeAllFollowButts.js

```diff
@@ -117,15 +122,16 @@ function fetchButt(butt, buttInfo) {
   dataRequester.send();
 }
 
-function addButtClickHandle(response, butt) {
+function addButtClickHandles(response, buttons) {
   // currently lacking error handling
-  var buttInfo = JSON.parse(butt.dataset.info);
-  assignInitialButtResponse(response, butt);
-  butt.onclick = function (e) {
-    e.preventDefault();
-    handleOptimisticButtRender(butt);
-  };
-  butt.dataset.clickInitialized = 'true';
+  buttons.forEach((butt) => {
+    assignInitialButtResponse(response, butt);
+    butt.onclick = function (e) {
+      e.preventDefault();
+      handleOptimisticButtRender(butt);
+    };
```

Comment:
I wonder if we can replace this with `addEventListener` or even with a single listener we can bubble up to, this way we won't need to assign a listener to each button

Reply:
The container can definitely handle the click for all the buttons. As well, the current click is not using addEventListener. When the container is clicked, you can check if the target or closest selector for `.follow-user` was clicked and handle it from there.

Reply:
The only thing I'm not sure about with attaching to the container is that there are multiple containers that have follow buttons (e.g. the sticky right-hand nav, the author byline, liquid tags, profile page header)... I guess the click handler would need to be attached to the `body` element itself for us to handle them all in one like that

Reply:
I think you could have a body click that handles this or if you're certain of where the follow buttons are and want to be more laser focused, it could be one click event handler per follow button container. If we're certain of the selector though, it's probably fine right on the body.

Reply:
I'll have a look around and see but my instinct is these follow buttons are _everywhere_ and the regression risk in attaching a handler per container might be quite high (I'm bound to miss a container somewhere 😅 ). 

I feel like the ideal solution might be changing this from an initializer to a helper called from page-specific packs, which could target the containers specific to those pages... but I think the chances of regression would make that more suited to a separate PR than this bugfix around network requests.

Reply:
Sounds good!